cryodrgn: add v3.4.3

snehring opened the pull request:
I debated adding in a +cuda variant and depending on py-torch+cuda when that's specified, but you still need to specify cuda_arch for py-torch in that scenario. I think I just need to accept that if you want this installed with gpu acceleration for torch you better tell it to use a torch with gpu support enabled.

snehring committed:
cryodrgn: adding v3.4.3

Signed-off-by: Shane Nehring <snehring@iastate.edu>

RikkiButler20 commented:
Can you add conditionals to the updated dependencies? It seems that the changes only apply to v3.4.3 based on cryodrgn’s pyproject.toml
[Spack's conditional dependencies](https://spack.readthedocs.io/en/latest/packaging_guide.html#conditional-dependencies)
[Cryodrgn pyproject.toml v3.4.3](https://github.com/ml-struct-bio/cryodrgn/blob/3.4.3/pyproject.toml)

snehring commented:
Added in a separate python dep for 2.3.0 to allow for 3.7-3.11 and a conditional on the py-notebook, which doesn't seem to be needed for 2.3.0. Rebased to tip of develop.

RikkiButler20 reviewed (approved):
LGTM
